test/CMakeLists.txt

```
@@ -10,15 +10,19 @@ add_subdirectory(v2gtp)
 # code coverage specifics
 #
 
-evc_include(CodeCoverage)
-append_coverage_compiler_flags_to_target(iso15118)
+if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU" OR CMAKE_C_COMPILER_ID STREQUAL "GNU")
+    evc_include(CodeCoverage)
+    append_coverage_compiler_flags_to_target(iso15118)
 
-setup_target_for_coverage_gcovr_html(
-    NAME ${PROJECT_NAME}_gcovr_coverage
-    EXECUTABLE ctest
-)
+    setup_target_for_coverage_gcovr_html(
+        NAME ${PROJECT_NAME}_gcovr_coverage
+        EXECUTABLE ctest
+    )
 
-setup_target_for_coverage_gcovr_xml(
-    NAME ${PROJECT_NAME}_gcovr_coverage_xml
-    EXECUTABLE ctest
-)
+    setup_target_for_coverage_gcovr_xml(
+        NAME ${PROJECT_NAME}_gcovr_coverage_xml
+        EXECUTABLE ctest
+    )
+else()
+    message(WARNING "Code coverage is disabled when building with clang")
```

Maybe reword this so it doesn't single-out clang, since it is only enabled when building with gcc
```suggestion
    message(WARNING "Code coverage is disabled when not building with gcc")
```